core/web-assets/src/main/assets/js/apps/onms-classifications/index.js

```diff
@@ -312,6 +314,59 @@ const confirmTopoverTemplate = require('./views/modals/popover.html');
                 });
             };
 
+            // for drag and drop of rules (redefining position)
+            $scope.sortableRules = {
+
+                start: function(e, ui) {
+                    // remember old index before moving
+                    $(ui.item).data('oldIndex', ui.item.index());
+                },
+                stop: function(e, ui) {
+
+                    // 1.) Check Precondition: not read only group
+                    if($scope.group.readOnly) {
+                        ui.item.parent().sortable('cancel');
+                        $window.alert('this group cannot be sorted, it is read only');
+                        return;
+                    }
+
+                    // 2.) Check Precondition: must be sorted by position
+                    if(!($scope.query.orderBy === 'position' && $scope.query.order === 'asc')) {
+                        ui.item.parent().sortable('cancel');
+                        $window.alert('You can only reorder the position if sorted by position');
+                        return;
+                    }
+
+                    // 3.) Check Precondition: item was actually moved
+                    var oldIndex =  $(ui.item).data().oldIndex;
+                    var newIndex =  ui.item.index();
+                    if(oldIndex === newIndex) {
+                        ui.item.parent().sortable('cancel');
+                        return; // nothing to do
+                    }
+
+                    // 4.) calculate and set new position (index + offset)
+                    var parameters = $scope.query || {};
+                    var offset =  (parameters.page -1) * parameters.limit || 0;
+                    var rule = $scope.rules[newIndex];
+                    rule.position = newIndex + offset;
+
+                    // 5.) update backend
+                    var refreshCallback = function() {
+                        $scope.refreshAll();
+                    };
+
+                    var handleErrorResponse = function(response) {
```

Comment:
Can we make this a first level function and re-use where needed instead of duplicating it.

Reply:
I am not sure how to do this? How would that look like?